test/daoreg.token.test.js

```diff
@@ -58,133 +56,735 @@ describe('Tests for tokens in dao registry', async function () {
+  })
 
+  it('Token maximum supply can not be greater than 1e15', async function(){
   	// Arrange
+    let fail 
+    let error
     const [token_contract, token_account] = await TokenUtil.createTokenContract();
+    // Act
+
+    try{
+      await TokenUtil.createWithErrors({ 
+        issuer: daoreg, 
+        maxSupply: `1000000000000000.0000 ${TokenUtil.tokenTest}`,
+        contractAccount: token_account,
+        contract: token_contract
+      })  
+      fail = false
+    }catch(err){
+      fail = true
+      error = err
+    }
+
+    //Assert
+    expect(fail).to.be.true
+  })
 
+  it('Token max-supply must be positive', async function () {
+  	// Arrange
+    let fail 
+    let error
+    const [token_contract, token_account] = await TokenUtil.createTokenContract();
+    
     // Act
-    await TokenUtil.create({ 
+    try{
+      await TokenUtil.createWithErrors({ 
+        issuer: daoreg, 
+        maxSupply: `-10000000000000.0000 ${TokenUtil.tokenTest}`,
+        contractAccount: token_account,
+        contract: token_contract
+      })  
+      fail = false
+    }catch(err){
+      fail = true
+      error = err
+    }
+    //Assert
+    expect(fail).to.be.true
+  })
+
+  it('Token symbol already exists', async function () {
+  	// Arrange
+    let fail
+    let error
+    const [token_contract, token_account] = await TokenUtil.createTokenContract();
+    
+    // Act
+    try{
+      await TokenUtil.createWithErrors({ 
+        issuer: daoreg, 
+        maxSupply: `10000.0000 ${TokenUtil.tokenTest}`,
+        contractAccount: token_account,
+        contract: token_contract
+      })  
+      await TokenUtil.createWithErrors({ 
+        issuer: daoreg, 
+        maxSupply: `10000.0000 ${TokenUtil.tokenTest}`,
+        contractAccount: token_account,
+        contract: token_contract
+      })
+      fail = false
+    }catch(err){
+      fail = true
+      error = err
+    }
+    //Assert
+    expect(fail).to.be.true
+
+  })
+
+  it('Issue the new token', async function () {
+    // Arrange
+    const [token_contract, token_account] = await TokenUtil.createTokenContract();
+    await TokenUtil.createWithErrors({ 
       issuer: daoreg, 
-      maxSupply: `1000000000000.0000 DTK`,
+      maxSupply: `10000.0000 ${TokenUtil.tokenTest}`,
       contractAccount: token_account,
       contract: token_contract
     })
 
+    // // Act
+    await TokenUtil.issue({
+      supply: `4000.0000 ${TokenUtil.tokenTest}`,
+      issuer: daoreg,
+      contract: token_contract,
+      memo: 'issued token'
+    })
+
     // Assert
+    
+    let ret = await rpc.get_currency_stats(token_account, TokenUtil.tokenTest)
+    
+    expect(ret[`${TokenUtil.tokenTest}`]).to.deep.equals({
+      supply: `4000.0000 ${TokenUtil.tokenTest}`,
+      max_supply: `10000.0000 ${TokenUtil.tokenTest}`,
+      issuer: daoreg
+      })
 
   })
 
-  it('Issue the new token', async function () {
+  it('You can not issue, token symbol is not valid', async function () {
```

Comment:
Also, there is a lot of 'you' in the test.
I think it would look better if you call as actor as user
i.e. _user can not issue an invalid token_